lib/zlib.toit

```diff
@@ -8,23 +8,45 @@ import crypto
 import crypto.adler32 as crypto
 import crypto.crc as crc_algorithms
 
-class ZlibEncoder_ implements reader.Reader:
+class RleReader implements reader.Reader:
+  owner_ /ZlibEncoder_? := null
+
+  constructor.private_:
+
+  read:
+    return owner_.read_
+
+  close:
+    // Currently does nothing.
+
+class ZlibEncoder_ implements reader.Reader:  // TODO: Should not be a reader.
   channel_ := monitor.Channel 1
+  reader/RleReader ::= RleReader.private_
 
   static SMALL_BUFFER_DEFLATE_HEADER_ ::= [8, 0x1d]
   static MINIMAL_GZIP_HEADER_ ::= [0x1f, 0x8b, 8, 0, 0, 0, 0, 0, 0, 0xff]
 
   constructor --gzip_header/bool:
+    reader.owner_ = this
     header := gzip_header ? MINIMAL_GZIP_HEADER_ : SMALL_BUFFER_DEFLATE_HEADER_
     channel_.send (ByteArray header.size: header[it])
 
+  /**
+  Deprecated.
+  Use the $reader to get an object with a read method.
```

Comment:
Add comment that this should also remove the `implements reader.Reader` when the method is removed.